src/notification/extract.js

```diff
@@ -172,13 +156,19 @@ export const fromAPNsImpl = (rawData: ?JSONableDict): Notification | void => {
     throw err('invalid');
   }
 
-  const { realm_uri } = zulip;
+  const { realm_uri, user_id } = zulip;
   if (realm_uri === undefined) {
     throw err('archaic (pre-1.9.x)');
   }
   if (typeof realm_uri !== 'string') {
     throw err('invalid');
   }
+  if (user_id === undefined) {
+    throw err('archaic (pre-2.1.x)');
```

Comment:
This should get a test case. See the existing mentions of `archaic` in the tests.

... Which raises the question why this "baseline" case didn't break:
```js
      // baseline
      expect(make({ realm_uri, recipient_type: 'private', sender_email })).toBeTruthy();
      // missing recipient_type
      expect(make({ realm_uri, sender_email })).toThrow(/archaic/);
```
It looks like the answer is that it's passing because the `make` return value is just a closure, and so always truthy -- it needs to be called to actually test anything. Oops. So a prep commit should fix that.